packages/preview/xyznote/0.2.0/template.typ

```diff
@@ -0,0 +1,265 @@
+    #abstract
+  ]
+
+  align(bottom + center)[
+    #set text(size: 15pt)
+    *#author*
+  ]
+
+  align(bottom + center)[
+    #set text(size: 15pt)
+    *#createtime*
+  ]
+
+
+
+  //页眉
+  set page(header: context {
+    set text(font: ("Libertinus Serif", "NSimSun"))
+    if here().page() == 1 {
+      return
+    }
+
+    let elems = query(heading.where(level: 1).after(here()))
+
+    let chapter-title = ""
+
+    if (elems == () or elems.first().location().page() != here().page()) {
+      let elems = query(heading.where(level: 1).before(here()))
+      chapter-title = elems.last().body
+    } else {
+      chapter-title = elems.first().body
+    }
+    align(right)[#chapter-title]
+
+    v(-8pt)
+    align(center)[#line(length: 105%, stroke: (thickness: 1pt, dash: "solid"))]
+  })
+
+
+
+  //引用块
+  set quote(block: true)
+
+
+
+  // // codeblock
+  // show raw.where(block: true): block.with(
+  //   fill: luma(87.45%),   // 设置背景颜色
+  //   inset: 7pt,         // 设置内边距
+  //   radius: 2pt,         // 设置圆角半径
+  //   width: 100%,         // 设置宽度为 100%
+  // )
+  //inlinecode
+  show raw.where(block: false): it => box(fill: rgb("#d7d7d7"), inset: (x: 2pt), outset: (y: 3pt), radius: 1pt)[#it]
+
+  // show raw: set text(font: "jetbrains mono") //修改代码字体
+
+  show table.cell.where(y: 0): strong //表格表头加粗
+
+  //链接下划线
+  show link: {
+    underline.with(stroke: blue, offset: 2pt)
+  }
+
+  //外部包
+  import "@preview/codly:1.0.0": *
+  import "@preview/codly-languages:0.1.1": *
+  show: codly-init //初始化 codly
+  // codly(number-format: none) //不显示行号
+  codly(languages: codly-languages) //设置语言图标
+
+
+  //公式编号
+  set math.equation(numbering: (..nums) => (
+    context {
+      set text(size: 9pt)
+      numbering("(1.1)", chaptercounter.at(here()).first(), ..nums)
+    }
+  ))
+
+  // caption 计数器
+  set figure(numbering: (..nums) => (
+    context {
+      set text(font: ("Libertinus Serif", "KaiTi"), size: 9pt)
+      numbering("1.1", chaptercounter.at(here()).first(), ..nums)
+    }
+  ))
+
+  //图片表格 caption 字体字号
+  show figure.caption: set text(font: ("Libertinus Serif", "KaiTi"), size: 9pt)
+
+  // ------------------------ 以下为正文配置 ---------------------------------- //
+
+  //正文页边距
+  set page(margin: (
+    top: 2cm,
+    bottom: 2cm,
+    right: 2cm,
+    left: 2cm,
+  ))
+
+  //正文字体字号
+  set text(
+    font: ("Libertinus Serif", "microsoft yahei"),
+    // size: 11pt,
+  )
+
+  // set text(tracking: 1pt) //字间距
+  // set par(leading: 1pt) //行间距
+
+  //段落缩进
+  // set par(
+  //   justify: true,
+  //   first-line-indent: 1em,
+  // )
+
+  pagebreak()
+
+  outline()
+
+  pagebreak()
+
+  //页码
+  set page(
+    numbering: "1 / 1",
+    number-align: right,
+  )
+  counter(page).update(1)
+
+  body
+
+  //参考文献
+  if bibliography-file != none {
+    set text(font: ("Times New Roman", "KaiTi")) //设置参考文献字体
+    pagebreak()
+    show bibliography: set text(10.5pt)
+    bibliography("template/refs.bib", style: bibliographystyle)
```

Comment:
Users of your package won't be able to give their own bibliography file with that code. The `refs.bib` file in your package and in their project are not the same, and here you are referencing the one from the package, which people can't edit.
The proper approach here is to let users pass a `bibliography(...)` call as a parameter to your template, that you simply display.

```typst
#let xyznote(
  // ...
  bib: none
) {
  // ...
  if bib != none {
    // Add the required configuration as before
    set text(font: ("Times New Roman", "KaiTi"))
    pagebreak()
    show bibliography: set text(10.5pt)
    // Set the bibliography style with a set rule
    set bibliography(style: bibliographystyle)
    // Show the bibliography the user supplied
    #bib
  }
}
```

```typst
#show: xyznote.with(bib: bibliography("my-refs.bib"))
```

Reply:
Okay, I'll make some more changes.